openfl-tutorials/interactive_api/PyTorch_Huggingface_transformers_SUPERB/workspace/PyTorch_Huggingface_transformers_SUPERB.ipynb

```diff
@@ -0,0 +1,485 @@
+    "\n",
+    "optimizer = AdamW(params_to_update, lr=3e-5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "caroline-passion",
+   "metadata": {},
+   "source": [
+    "#### Register model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "handled-teens",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "framework_adapter = (\n",
+    "    \"openfl.plugins.frameworks_adapters.pytorch_adapter.FrameworkAdapterPlugin\"\n",
+    ")\n",
+    "MI = ModelInterface(\n",
+    "    model=model, optimizer=optimizer, framework_plugin=framework_adapter\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "portuguese-groove",
+   "metadata": {},
+   "source": [
+    "### Define and register FL tasks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5d4ff313-a17f-4119-a4c7-afa898b0f304",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "batch_size = 16\n",
+    "args = TrainingArguments(\n",
+    "    \"finetuned_model\",\n",
+    "    save_strategy=\"epoch\",\n",
+    "    per_device_train_batch_size=batch_size,\n",
+    "    per_device_eval_batch_size=batch_size,\n",
+    "    num_train_epochs=1,\n",
+    "    warmup_ratio=0.1,\n",
+    "    logging_steps=10,\n",
+    "    push_to_hub=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "increasing-builder",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TI = TaskInterface()\n",
+    "\n",
+    "import torch.nn as nn\n",
+    "import tqdm\n",
+    "\n",
+    "\n",
+    "@TI.register_fl_task(\n",
+    "    model=\"model\", data_loader=\"train_loader\", device=\"device\", optimizer=\"optimizer\"\n",
+    ")\n",
+    "def train(model, train_loader, optimizer, device):\n",
+    "\n",
+    "    print(f\"\\n\\n TASK TRAIN GOT DEVICE {device}\\n\\n\")\n",
+    "\n",
+    "    model = nn.DataParallel(model, device_ids=[int(device.split(\":\")[1])])\n",
+    "    trainer = Trainer(\n",
+    "        model,\n",
+    "        args,\n",
+    "        train_dataset=train_loader,\n",
+    "        tokenizer=feature_extractor,\n",
+    "        optimizers=(optimizer, None),\n",
+    "        compute_metrics=compute_metrics,\n",
+    "    )\n",
+    "    train_metrics = trainer.train()\n",
+    "    return {\"train_loss\": train_metrics.metrics[\"train_loss\"]}\n",
+    "\n",
+    "\n",
+    "@TI.register_fl_task(model=\"model\", data_loader=\"val_loader\", device=\"device\")\n",
+    "def validate(model, val_loader, device):\n",
+    "\n",
+    "    print(f\"\\n\\n TASK VALIDATE GOT DEVICE {device}\\n\\n\")\n",
+    "\n",
+    "    model = nn.DataParallel(model, device_ids=[int(device.split(\":\")[1])])\n",
+    "    trainer = Trainer(\n",
+    "        model,\n",
+    "        args,\n",
+    "        eval_dataset=val_loader,\n",
+    "        tokenizer=feature_extractor,\n",
+    "        compute_metrics=compute_metrics,\n",
```

Comment:
Could you please define `compute_metrics` before using this. I understand that it's working correct due to late binding but for users it would be easier to understand the tutorial.